### ISSUE TITLE
Firefox: this.endNode is null

### ISSUE THREAD

**Issue by NeoLegends** [Polymer/lit-html#264]
Hey!

I've just been adapting `fit-html` to the new Shady DOM implementation (which has been working fine so far), but I noticed a crash that affects (at least) Firefox:

Inserting a NodePart sometimes throws because the node's `endPart` is null. https://github.com/Polymer/lit-html/blob/c82ed698987b29d48570a3dc960f10fd33378b06/src/lit-html.ts#L556

The issue does not occur in Chrome or Safari. So it might have something to do with ShadyDOM or with Firefox in general. I was able to track down the bug via https://github.com/Polymer/lit-html/blob/c82ed698987b29d48570a3dc960f10fd33378b06/src/lit-html.ts#L525 and https://github.com/Polymer/lit-html/blob/c82ed698987b29d48570a3dc960f10fd33378b06/src/lit-html.ts#L683 to here https://github.com/Polymer/lit-html/blob/c82ed698987b29d48570a3dc960f10fd33378b06/src/lit-html.ts#L741

As you can see, lit passes `walker.currentNode.nextSibling` as `endNode` into the NodePart. In Chrome, this is never null / falsey, while in Firefox, it sometimes is. I wonder why. :D

**Comment by NeoLegends** [comment]
Uh, this seems to be same as https://github.com/Polymer/lit-html/issues/231. Just this time with official Shady DOM support.

**Comment by NeoLegends** [comment]
Testing shows that this is resolved by https://github.com/Polymer/lit-html/pull/241.

**Comment by justinfagnani** [comment]
Awesome, thanks for checking!

**Comment by NeoLegends** [comment]
My pleasure. Would be nice to also add a test covering this.

**Comment by justinfagnani** [comment]
Definitely. I filed https://github.com/Polymer/lit-html/issues/331 for that.
